src/bar_chart.ts

```diff
@@ -122,44 +136,97 @@ class BarChart implements IAppView {
     this.$node.selectAll(`div.bars > .${changeType.type}`).classed('hidden', !changeType.isActive);
   }
 
-  private requestData() {
+  private widthBar = 15;
+
+  private windowResize(items) {
+    this.items = items;
+    let width = $(window).innerWidth();
+
+    this.generateIndexArray(this.items, width);
+
+    const elements = d3.selectAll('.bars');
+    if (elements.empty() === false) {
+
+      this.requestData(width, this.leftValue);
+      elements.remove();
+    }
+    this.leftValue = [];
+
+  }
+
+  private requestData(totalWidth, leftValue) {
     return d3.pairs(this.items)
       .map((pair) => {
-        let ids = pair.map((d: any) => d.item.desc.id);
+        let ids = pair.map((d:any) => d.item.desc.id);
         return Promise.all([ajax.getAPIJSON(BarChart.getURL(ids)), pair, ids])
           .then((args) => {
             const json = args[0];
             const pair = args[1];
             const ids = args[2];
-            this.drawBars(json, pair, ids);
+
+            this.drawBars(json, pair, ids, leftValue.shift(), totalWidth);
+
           });
       });
   }
 
-  private drawBars(data, pair, ids) {
+  private drawBars(data, pair, ids, circleScale, totalWidth) {
+
     const that = this;
+    const posXScale = getPosXScale(this.items, totalWidth);
+
+    const posX = posXScale(moment(pair[0].time).diff(moment(this.items[0].time), 'days'))
+      + 0.5 * (posXScale(moment(pair[1].time).diff(moment(this.items[0].time), 'days'))
+      - posXScale(moment(pair[0].time).diff(moment(this.items[0].time), 'days')));
 
-    const posXScale = getPosXScale(this.items, this.totalWidth);
-    const posX = posXScale(moment(pair[0].time).diff(moment(this.items[0].time), 'days'));
+    const w = 80;
```

Comment:
I think I can´t put the constants out of this function because I use "pair" and "totalWidth" as transfer parameters for the calculation of these values??